core/trino-main/src/test/resources/catalog_impersonation.json

```diff
@@ -21,6 +21,12 @@
         {
             "original_user": ".*",
             "new_user": "test"
+        },
+        {
+            "original_user": "canImpersonateAnyUser"
```

Comment:
Why not to use `"new_user": ".*"`? This is a default so it should just work. What am I missing?

Reply:
My goal was to test that I don't need to specify `new_user` at all - i should be implicit `.*`.
Should i add another testcase that tests `"new_user": ".*"` explicitly?

Reply:
The `"new_user": ".*"` part is already tested with a test-case above
```
        {
            "original_user": "admin(-.*)?",
            "new_user": ".*"
        },
```

Reply:
> This is a default so it should just work.

Oh. I was wrong. I misread the diff previously. Sorry about it. It is your change which makes it optional with a default.

What is wrong with explicit: `"new_user": ".*"`?. Notice that having this optional sounds a bit risky as in case of misconfiguration we could escalate the access while now we fail.

Reply:
There is nothing wrong with the explicit `new_user` attribute. I just noticed that all the rules only have the `allow/owner/privileges` attribute required. So you can specify the resulting value and any other attribute you want to filter on.
Only the impersonation rule differs:
* `allow` is optional and defaulting to true
* `new_user` is required

Follow the style of all other rules `allow` should be mandatory and `new_user` should be optional IMHO.
Changing `allow` to required would be a breaking change for existing rules so i would not change it. But for the `new_user` part i would suggest to set it to optional too match all the other rules. This change is backward-compatible. I see the risk of not setting the attribute `new_user`, but this applies to all other rules as well.

Reply:
> Only the impersonation rule differs

That is true. That is because impersonation itself is different. Notice that if you would have by default `new_user=.*` then basically the entire security model is compromised. 

Notice that `allow` is optional here because `new_user` is required.

I would suggest to leave this as it is (abandon this change). Maybe instead we should add a comment why it was implemented this way, maybe even we could mention this in docs.

Other than that first 2 commits are good. Thanks!

Reply:
That would be fine for me, i have no strong opinion on this. I would be great if you could help my out with the following doc text, i will add it to the docs then.

The Impersonation rules are a bit different than the other rules: The attribute new_user is required to not accidentally prevent more access than intended. Doing so it was possible to make the attribute allow optional.

Reply:
It looks ok to me.

@mosabua ?